haskell-lsp-types/src/Language/Haskell/LSP/Types/Uri.hs

```diff
@@ -22,17 +22,27 @@ instance NFData Uri
 -- | When URIs are supposed to be used as keys, it is important to normalize
 -- the percent encoding in the URI since URIs that only differ
 -- when it comes to the percent-encoding should be treated as equivalent.
-newtype NormalizedUri = NormalizedUri Text
-  deriving (Eq,Ord,Read,Show,Generic,Hashable)
+--
+-- If you care about performance then you should use a hash map. The keys
+-- are cached in order to make hashing very fast.
+data NormalizedUri = NormalizedUri Int !Text
```

Comment:
Is the hash intentionally non-strict? I guess it saves you a bit if you don’t care about it but if you do care about it, making it strict so it can be unboxed seems benefitiial.

Reply:
You want to delay the hashing until it is actually used, which may never happen. So it makes sense to keep it non-strict

Reply:
The hash is intentionally non-strict for the reasons that Pepe suggests.